src/handler/sign_tx.c

```diff
@@ -23,13 +23,39 @@
 #include "os.h"
 #include "cx.h"
 #include "buffer.h"
+#include "swap.h"
 
 #include "sign_tx.h"
 #include "sw.h"
 #include "globals.h"
 #include "display.h"
 #include "tx_types.h"
 #include "deserialize.h"
+#include "handle_swap.h"
+#include "validate.h"
+
+#ifdef HAVE_SWAP
+static int sign_swap_tx(transaction_t *tx) {
+    if (G_swap_response_ready) {
+        // Safety against trying to make the app sign multiple TX
+        // This code should never be triggered as the app is supposed to exit after
+        // sending the signed transaction
+        PRINTF("Safety against double signing triggered\n");
+        os_sched_exit(-1);
+    } else {
+        // We will quit the app after this transaction, whether it succeeds or fails
+        PRINTF("Swap response is ready, the app will quit after the next send\n");
+        // This boolean will make the io_send_sw family instant reply +
+        // return to exchange
+        G_swap_response_ready = true;
+    }
+    if (swap_check_validity(tx->value, tx->fee, tx->to)) {
+        PRINTF("Swap response validated\n");
+        validate_transaction(true);
+    }
+    return 0;
```

Comment:
Add a comment like
`// Unreachable because swap_check_validity returns an error to exchange OR validate_transaction returns a success to exchange`
We don't want external devs to be confused about the sequence flow after signing / error